src/bus.rs

```diff
@@ -2,10 +2,24 @@ use core::slice;
 
 use embassy_time::{Duration, Timer};
 use embedded_hal_1::digital::OutputPin;
-use embedded_hal_async::spi::{transaction, SpiBusRead, SpiBusWrite, SpiDevice};
+use embedded_hal_1::spi::ErrorType;
+use embedded_hal_async::spi::{transaction, SpiDevice};
+use futures::FutureExt;
 
 use crate::consts::*;
 
+/// Custom Spi Trait that _only_ supports the bus operation of the cyw43
```

Comment:
It's a bit strange to use the standard `embedded-hal` `SpiDevice` trait with a custom non-`embedded-hal`. bus trait. Plus, it won't play nice with this proposed trait change: https://github.com/rust-embedded/embedded-hal/pull/443.

I think it makes sense that if we're going to do a higher-level "cyw43 bus" trait, it should be fully high-level, handling CS itself.

i.e. instead of requiring `where SPI: SpiDevice, SPI::Bus: SpiBusCyw43`, require `where SPI: SpiCyw43`. And then, make the impl toggle CS instead of relying on `SpiDevice`.

Reply:
That makes sense. I meant to be "minimally invasive" but i agree that the embedded-hal traits don't do a lot here. I will remove it.